src/ray/raylet/node_manager.cc

```diff
@@ -261,6 +261,35 @@ void NodeManager::ProcessNewClient(std::shared_ptr<LocalClientConnection> client
   client->ProcessMessages();
 }
 
+void NodeManager::DispatchTasks() {
+  // Work with a copy of scheduled tasks.
+  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  // Return if there are no tasks to schedule.
+  if (scheduled_tasks.empty()) {
+    return;
+  }
+  // Early return if there are no resources available.
+  const ClientID &my_client_id = gcs_client_->client_table().GetLocalClientId();
+  const auto &local_resources =
+      cluster_resource_map_[my_client_id].GetAvailableResources();
+  if (local_resources.IsEmpty()) {
+    return;
+  }
+
+  for (const auto &task : scheduled_tasks) {
+    const auto &task_resources = task.GetTaskSpecification().GetRequiredResources();
+    if (!task_resources.IsSubset(local_resources)) {
+      // Not enough local resources for this task right now, skip this task.
+      continue;
+    }
+    // We have enough resources for this task. Assign task.
+    // TODO(atumanov): perform the task state/queue transition inside AssignTask.
+    auto scheduled_tasks =
```

Comment:
Can we call this something other than `scheduled_tasks`? It's a little confusing given the other `scheduled_tasks` local variable here.